specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json

```diff
@@ -0,0 +1,646 @@
+      "get": {
+        "tags": [
+          "DatabaseVulnerabilityAssessmentScans"
+        ],
+        "description": "Gets a vulnerability assessment scan record of a database.",
+        "operationId": "DatabaseVulnerabilityAssessmentScans_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "name": "vulnerabilityAssessmentName",
+            "in": "path",
+            "description": "The name of the vulnerability assessment.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "default"
+            ],
+            "x-ms-enum": {
+              "name": "VulnerabilityAssessmentName",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "scanId",
+            "in": "path",
+            "description": "The vulnerability assessment scan Id of the scan to retrieve.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the database vulnerability assessment scan record.",
+            "schema": {
+              "$ref": "#/definitions/VulnerabilityAssessmentScanRecord"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidStorageAccountName - The provided storage account is not valid or does not exist.\n\n * 400 InvalidStorageAccountCredentials - The provided storage account shared access signature is not valid.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance.\n\n * 404 SourceDatabaseNotFound - The source database does not exist.\n\n * 409 DatabaseVulnerabilityAssessmentScanIsAlreadyInProgress - Vulnerability Assessment scan is already in progress.\n\n * 500 DatabaseIsUnavailable - Loading failed. Please try again later.\n\n * 500 DatabaseVulnerabilityAssessmentOperationFailed - An error has occurred while trying to execute Vulnerability Assessment operation, please try again later"
+          }
+        },
+        "x-ms-examples": {
+          "Gets a database vulnerability assessment scan record by scan ID": {
+            "$ref": "./examples/DatabaseVulnerabilityAssessmentScanRecordsGet.json"
+          }
+        }
+      },
+      "post": {
```

Comment:
Why is a scan not created (initiated) using PUT instead of POST?

Reply:
Put is traditionally used to create a resource. This API is for executing a scan (doing operation on a resource), so we thought POST will be a better match.

Reply:
@yaakoviyun - But this is not modeled as an action on a resource. Should it be modeled like - POST subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/vulnerabilityAssessments/{vulnerabilityAssessmentName}/initiateScan

Reply:
OK Will make the change in the next API-version of this API.

Reply:
We decided to implement it for this api version